hare/messagevalidation_test.go

```diff
@@ -65,17 +65,6 @@ func TestMessageValidator_ValidateCertificate(t *testing.T) {
 	cert.Values = valueSet.ToSlice()
 	require.False(t, sv.validateCertificate(context.Background(), cert))
 
-	msgs = make([]Message, 0, sv.threshold)
-	for i := 0; i < sv.threshold; i++ {
-		signer, err := signing.NewEdSigner()
-		require.NoError(t, err)
-		// wrong set value
-		m := BuildCommitMsg(signer, NewDefaultEmptySet()).Message
-		msgs = append(msgs, m)
-	}
-	cert.AggMsgs.Messages = msgs
-	require.False(t, sv.validateCertificate(context.Background(), cert))
```

Comment:
In the unchanged code this part of the test passes because `func (v *syntaxContextValidator) validateAggregatedMessage` returns `false` do to the hash in the Message not matching the actual hash of the message, not because the set value is wrong.

Since this check isn't done any more the `sv.validateCertificate` returns `true` here.

I think the previous intention of the test was to verify that invalid sets fail validation, but according to another test an empty set isn't invalid?

Reply:
As discussed with @countvonzero the actual check here was to see if the message has been altered after signing.

With the change the NodeID extracted from the signature will differ from the ID of the smesher that signed the message.

This is detected in the eligibility oracle where the check for the NodeID being in the active set is now done first so that the check fails faster if this happens.